InfiniTAM/InfiniTAM_node.cpp

```diff
@@ -169,24 +184,24 @@ int main(int argc, char** argv) try {
   }
 
   printf("initialising ...\n");
-  ImageSourceEngine* image_source = NULL;
-  IMUSourceEngine* imu_source = NULL;
 
-  CreateDefaultImageSource(arg1, arg2, arg3, arg4, node_handle, image_source,
-                           imu_source);
+  SetUpSources(arg1, arg2, arg3, arg4, node_handle, image_source, imu_source);
```

Comment:
I'm not passing the main_engine because main_engine has to be initialized after the initialization of image_source (main_engine gets an instance of image_source).
And in image_source I need to access the main_engine, its a chicken or the egg dilemma.